From 611b5158a89427e45bedf2703b7430da5d8156dc Mon Sep 17 00:00:00 2001
From: Andrew Bayer <andrew.bayer@gmail.com>
Date: Sat, 5 Feb 2011 16:04:34 -0800
Subject: [PATCH 1/7] Applying CLOUDERA-BUILD commits to cdh-1.3.0:

 CLOUDERA-BUILD. Adding build infrastructure.
 CLOUDERA-BUILD. Improvements to install_sqoop.sh
   Add sqoop-job, sqoop-metastore manpages and scripts to installation.
   Add doc/ files and subdirs to installation.
   Set /usr/share/man/man1 as correct default manpage dir.
   Remove .gitignore from etc/sqoop
 CLOUDERA-BUILD. Add sqoop-merge to the install list.
 CLOUDERA-BUILD. Changes to support CDH Mavenization.
 CLOUDERA-BUILD. Commenting out shims from install_sqoop.sh since
   they no longer exist in 1.2.0.
 CLOUDERA-BUILD. Restricting classpath to jars.
 CLOUDERA-BUILD. Forcing use of Checkstyle 5.0
 CLOUDERA-BUILD. Preparing for CDH3B4 release.
 CLOUDERA-BUILD. Preparing for cdh3u0 development.
 CLOUDERA-BUILD. Removing inaccurate HBase dependency.
 CLOUDERA-BUILD. Fixing KITCHEN-815.
   * Now runs mvn before anything else happens (assuming cdh.maven is not
     set), thanks to ant-contrib.
 CLOUDERA-BUILD. Removing cleanup of build.properties for now.
 CLOUDERA-BUILD. CDH-2773 - use cdh.build.properties rather than build.properties.
 CLOUDERA-BUILD. Adding relativePath to cloudera-pom.xml.
 CLOUDERA-BUILD. Simplifying repository setup.
 CLOUDERA-BUILD. Generating source jar, using original actual jar rather than recreating.
 CLOUDERA-BUILD. Changing releases repo to point to staging area.
 CLOUDERA-BUILD. Updating versions for cdh3u0 release.
 CLOUDERA-BUILD. Attaching sqoop-test.jar as Maven artifact.
 CLOUDERA-BUILD. Updating for CDH3u1 development.
 CLOUDERA-BUILD. Adding avro dependency version.
   Sqoop 1.3.0 has dependency on Avro 1.5.1 which was not present in the prior
   release version.
---
 build.xml                                          |   47 +++--
 cloudera-pom.xml                                   |  238 ++++++++++++++++++++
 cloudera/README.cloudera                           |    9 +
 cloudera/do-release-build                          |   28 +++
 cloudera/install_sqoop.sh                          |  140 ++++++++++++
 cloudera/maven-packaging/pom.xml                   |   86 +++++++
 cloudera/maven-packaging/sqoop/pom.xml             |  149 ++++++++++++
 cloudera/maven-packaging/templates/build.template  |    8 +
 .../maven-packaging/templates/libraries.template   |   47 ++++
 ivy.xml                                            |   16 ++-
 ivy/ivysettings.xml                                |   24 ++-
 11 files changed, 769 insertions(+), 23 deletions(-)
 create mode 100644 cloudera-pom.xml
 create mode 100644 cloudera/README.cloudera
 create mode 100755 cloudera/do-release-build
 create mode 100755 cloudera/install_sqoop.sh
 create mode 100644 cloudera/maven-packaging/pom.xml
 create mode 100644 cloudera/maven-packaging/sqoop/pom.xml
 create mode 100644 cloudera/maven-packaging/templates/build.template
 create mode 100644 cloudera/maven-packaging/templates/libraries.template

diff --git a/build.xml b/build.xml
index ca92924..844ee33 100644
--- a/build.xml
+++ b/build.xml
@@ -21,8 +21,35 @@
     xmlns:artifact="urn:maven-artifact-ant"
     xmlns:ivy="antlib:org.apache.ivy.ant">
 
+  <!-- load ant-contrib tasks to get the "if" task. -->
+  <taskdef resource="net/sf/antcontrib/antcontrib.properties">
+    <classpath>
+      <pathelement location="${basedir}/lib/ant-contrib-1.0b3.jar"/>
+    </classpath>
+  </taskdef>
+  
+  <!-- Initializes ivy/libraries.properties and build.properties with CDH versions -->
+  <if>
+    <and>
+      <not>
+        <isset property="cdh.maven" />
+      </not>
+      <not>
+        <available file="${basedir}/cdh.build.properties" />
+      </not>
+    </and>
+    <then>
+      <exec  executable="mvn" dir="${basedir}" failonerror="true">
+        <arg value="-f" />
+        <arg value="cloudera-pom.xml" />
+        <arg value="process-resources"/>
+      </exec>
+    </then>
+  </if>
+
   <!-- Load system-wide and project-wide default properties set by
        the user, to avoid needing to override with -D. -->
+  <property file="${basedir}/cdh.build.properties" />
   <property file="${user.home}/build.properties" />
   <property file="${basedir}/build.properties" />
 
@@ -115,7 +142,7 @@
 
   <!-- maven -->
   <property name="mvn.build.dir" value="${build.dir}/m2" />
-  <property name="mvn.repo" value="snapshots" />
+  <property name="mvn.repo" value="releases" />
   <property name="mvn.repo.id" value="cloudera.${mvn.repo}.repo" />
   <property name="mvn.deploy.url"
       value="https://repository.cloudera.com/content/repositories/${mvn.repo}"/>
@@ -714,21 +741,9 @@
     </jar>
   </target>
 
-  <target name="mvn-install" depends="init,mvn-prep,jar-all,srcjars"
+  <target name="mvn-install"
       description="Install Sqoop in local m2 repository">
-    <artifact:pom file="${sqoop.pom}" id="sqoop" />
-    <artifact:install file="${build.dir}/${dest.jar}">
-      <pom refid="sqoop" />
-      <attach file="${build.srcjar.dir}/${artifact.name}-sources.jar"
-          classifier="sources" />
-    </artifact:install>
-
-    <artifact:pom file="${sqooptest.pom}" id="sqoop-test" />
-    <artifact:install file="${build.dir}/${test.jar}">
-      <pom refid="sqoop-test" />
-      <attach file="${build.srcjar.dir}/${name}-test-${version}-sources.jar"
-          classifier="sources" />
-    </artifact:install>
+    <echo>To install Cloudera Sqoop jars into Maven cache, run "mvn -f cloudera-pom.xml install" from this directory.</echo>
   </target>
 
   <target name="mvn-deploy" depends="init,mvn-prep,jar-all,srcjars"
@@ -1000,7 +1015,7 @@
           pattern="${build.ivy.lib.dir}/${ivy.artifact.retrieve.pattern}"
           sync="true" />
         <ivy:cachepath pathid="${name}.hadooptest.classpath"
-          conf="clouderatest" />
+          conf="clouderatest" type="jar"/>
   </target>
 
   <!-- retrieve ivy-managed artifacts for checkstyle -->
diff --git a/cloudera-pom.xml b/cloudera-pom.xml
new file mode 100644
index 0000000..e2cb518
--- /dev/null
+++ b/cloudera-pom.xml
@@ -0,0 +1,238 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<project>
+  <modelVersion>4.0.0</modelVersion>
+  
+  <parent>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>sqoop-root</artifactId>
+    <version>3.0-u1-SNAPSHOT</version>
+    <relativePath>cloudera/maven-packaging/pom.xml</relativePath>
+  </parent>
+  
+  <groupId>com.cloudera.cdh</groupId>
+  <artifactId>sqoop-ant</artifactId>
+  <version>3.0-u1-SNAPSHOT</version>
+  <packaging>pom</packaging>
+  
+  <scm>
+    <connection>scm:git:git@git.sf.cloudera.com:sqoop.git</connection>
+    <url>git@git.sf.cloudera.com:sqoop.git</url>
+  </scm>
+  
+  <description>Sqoop ant</description>
+  <name>Sqoop ant</name>
+  
+  <licenses>
+    <license>
+      <name>The Apache Software License, Version 2.0</name>
+      <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+    </license>
+  </licenses>
+  
+  <organization>
+    <name>Cloudera</name>
+    <url>http://www.cloudera.com</url>
+  </organization>
+  
+  <profiles>
+    <!--To allow standalone project to fetch CDH artifacts -->
+    <!--Activate by default doing a negation of negation   -->
+    <!-- this trick allows CDH root to switch it off       -->
+    <profile>
+      <id>cdh.repos</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+        <property>
+          <name>use.cdh.repos</name>
+          <value>!false</value>
+        </property>
+      </activation>
+      <repositories>
+        <repository>
+          <id>cdh.releases.repo</id>
+          <url>https://repository.cloudera.com/content/group/cdh-releases-rcs</url>
+          <name>CDH Releases Repository</name>
+          <snapshots>
+            <enabled>false</enabled>
+          </snapshots>
+        </repository>
+        <repository>
+          <id>cdh.snapshots.repo</id>
+          <url>https://repository.cloudera.com/content/repositories/snapshots</url>
+          <name>CDH Snapshots Repository</name>
+          <snapshots>
+            <enabled>true</enabled>
+          </snapshots>
+        </repository>
+      </repositories>
+    </profile>
+    <!-- Similar logic to allow us to skip compilation/tests when run -->
+    <!-- via do-release-build or packaging scripts, since we're already -->
+    <!-- compiling via direct ant calls. -->
+    <profile>
+      <id>non.release.build</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+        <property>
+          <name>not.cdh.release.build</name>
+          <value>!false</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <inherited>false</inherited>
+            <executions>
+              <execution>
+                <id>clean</id>
+                <configuration>
+                  <target>
+                    <exec executable="${ant.cmd}" dir="${basedir}" failonerror="true">
+                      <arg value="clean"/>
+                      <arg value="-Dcdh.maven=true"/>
+                      <arg value="-Divy.cache.dir=${ivy.cache.dir}"/>
+                    </exec>
+                  </target>
+                </configuration>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <phase>clean</phase>
+              </execution>
+              <execution>
+                <id>compile</id>
+                <configuration>
+                  <target>
+                    <exec executable="${ant.cmd}" dir="${basedir}" failonerror="true">
+                      <arg value="jar-all"/>
+                      <arg value="-Dcdh.maven=true"/>
+                      <arg value="-Dreactor.repo=file://${maven.repo.local}"/>
+                      <arg value="-Divy.cache.dir=${ivy.cache.dir}"/>
+                    </exec>
+                  </target>
+                </configuration>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <phase>compile</phase>
+              </execution>
+              <execution>
+                <id>test</id>
+                <configuration>
+                  <target unless="${skipTests}">
+                    <exec executable="${ant.cmd}" dir="${basedir}" failonerror="true">
+                      <arg value="test"/>
+                      <arg value="-Dcdh.maven=true"/>
+                      <arg value="-Dreactor.repo=file://${maven.repo.local}"/>
+                      <arg value="-Divy.cache.dir=${ivy.cache.dir}"/>
+                    </exec>
+                  </target>
+                </configuration>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <phase>test</phase>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <!-- Profile that's only activated when we're calling as a release build, to run -->
+    <!-- with variables set *and* tar target -->
+    <profile>
+      <id>cdh.release.build</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+        <property>
+          <name>not.cdh.release.build</name>
+          <value>false</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <inherited>false</inherited>
+            <executions>
+              <execution>
+                <id>tar</id>
+                <configuration>
+                  <target>
+                    <property file="${basedir}/cdh.build.properties"/>
+                    <exec executable="${ant.cmd}" dir="${basedir}" failonerror="true">
+                      <arg value="clean"/>
+                      <arg value="tar"/>
+                      <arg value="-Dreactor.repo=file://${maven.repo.local}"/>
+                      <arg value="-Divy.cache.dir=${ivy.cache.dir}"/>
+                      <arg value="-propertyfile" />
+                      <arg value="cloudera/build.properties"/>
+                    </exec>
+                  </target>
+                </configuration>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <phase>compile</phase>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+  
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>com.cloudera.kitchen</groupId>
+        <artifactId>maven-deps-to-props-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>libraries.properties</id>
+            <configuration>
+              <templateFile>${basedir}/cloudera/maven-packaging/templates/libraries.template</templateFile>
+              <targetFile>${basedir}/ivy/libraries.properties</targetFile>
+            </configuration>
+            <goals>
+              <goal>generate</goal>
+            </goals>
+            <phase>process-resources</phase>
+          </execution>
+          <execution>
+            <id>build.properties</id>
+            <configuration>
+              <templateFile>${basedir}/cloudera/maven-packaging/templates/build.template</templateFile>
+              <targetFile>${basedir}/cdh.build.properties</targetFile>
+            </configuration>
+            <goals>
+              <goal>generate</goal>
+            </goals>
+            <phase>process-resources</phase>
+          </execution>
+        </executions>
+      </plugin>
+      
+    </plugins>
+  </build>
+  
+</project>
diff --git a/cloudera/README.cloudera b/cloudera/README.cloudera
new file mode 100644
index 0000000..8626e8f
--- /dev/null
+++ b/cloudera/README.cloudera
@@ -0,0 +1,9 @@
+This build was generated by Cloudera's build system in the following manner:
+
+1) The pristine open-source release tarball was unpacked
+
+2) The patches contained within the patches/ directory next to this README
+were applied using the apply-patches script. A complete log of these changes
+is also included in CHANGES.cloudera.txt.
+
+3) The project was built by running the do-release-build script in this directory.
diff --git a/cloudera/do-release-build b/cloudera/do-release-build
new file mode 100755
index 0000000..c0f3305
--- /dev/null
+++ b/cloudera/do-release-build
@@ -0,0 +1,28 @@
+#!/bin/bash
+# Copyright (c) 2009 Cloudera, inc
+#
+# Performs a release build
+
+set -ex
+
+JAVA32_HOME=${JAVA32_HOME:-$JAVA_HOME}
+
+# Check that JVMs support the right architectures
+# Passing -d with an architecture that's not supported results
+# in a non-zero exit code even with -version
+if ! $JAVA32_HOME/bin/java -d32 -version ; then
+  echo Your jvm in $JAVA32_HOME seems like it doesnt support 32-bit mode 1>&2
+  echo Please set JAVA32_HOME to point to a 32-bit JDK
+  exit 1
+fi
+
+# Do the build
+BIN_DIR=$(readlink -f $(dirname $0))
+RELEASE_DIR=$BIN_DIR/..
+
+# Change to cloudera/maven directory, compile, install
+# (and if called from CDH nightly build, deploy) artifacts into Maven repository
+cd $BIN_DIR/maven-packaging
+JAVA_HOME=$JAVA32_HOME mvn -Dnot.cdh.release.build=false install $DO_MAVEN_DEPLOY
+
+
diff --git a/cloudera/install_sqoop.sh b/cloudera/install_sqoop.sh
new file mode 100755
index 0000000..37d853b
--- /dev/null
+++ b/cloudera/install_sqoop.sh
@@ -0,0 +1,140 @@
+#!/bin/sh
+# Copyright 2009 Cloudera, inc.
+set -ex
+
+usage() {
+  echo "
+usage: $0 <options>
+  Required not-so-options:
+     --cloudera-source-dir=DIR   path to cloudera distribution files
+     --build-dir=DIR             path to sqoopdist.dir
+     --prefix=PREFIX             path to install into
+
+  Optional options:
+     --doc-dir=DIR               path to install docs into [/usr/share/doc/sqoop]
+     --lib-dir=DIR               path to install sqoop home [/usr/lib/sqoop]
+     --installed-lib-dir=DIR     path where lib-dir will end up on target system
+     --bin-dir=DIR               path to install bins [/usr/bin]
+     --examples-dir=DIR          path to install examples [doc-dir/examples]
+     ... [ see source for more similar options ]
+  "
+  exit 1
+}
+
+OPTS=$(getopt \
+  -n $0 \
+  -o '' \
+  -l 'cloudera-source-dir:' \
+  -l 'prefix:' \
+  -l 'doc-dir:' \
+  -l 'lib-dir:' \
+  -l 'installed-lib-dir:' \
+  -l 'bin-dir:' \
+  -l 'examples-dir:' \
+  -l 'build-dir:' -- "$@")
+
+if [ $? != 0 ] ; then
+    usage
+fi
+
+eval set -- "$OPTS"
+set -ex
+while true ; do
+    case "$1" in
+        --cloudera-source-dir)
+        CLOUDERA_SOURCE_DIR=$2 ; shift 2
+        ;;
+        --prefix)
+        PREFIX=$2 ; shift 2
+        ;;
+        --build-dir)
+        BUILD_DIR=$2 ; shift 2
+        ;;
+        --doc-dir)
+        DOC_DIR=$2 ; shift 2
+        ;;
+        --lib-dir)
+        LIB_DIR=$2 ; shift 2
+        ;;
+        --installed-lib-dir)
+        INSTALLED_LIB_DIR=$2 ; shift 2
+        ;;
+        --bin-dir)
+        BIN_DIR=$2 ; shift 2
+        ;;
+        --examples-dir)
+        EXAMPLES_DIR=$2 ; shift 2
+        ;;
+        --)
+        shift ; break
+        ;;
+        *)
+        echo "Unknown option: $1"
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+for var in CLOUDERA_SOURCE_DIR PREFIX BUILD_DIR ; do
+  if [ -z "$(eval "echo \$$var")" ]; then
+    echo Missing param: $var
+    usage
+  fi
+done
+
+DOC_DIR=${DOC_DIR:-/usr/share/doc/sqoop}
+LIB_DIR=${LIB_DIR:-/usr/lib/sqoop}
+BIN_DIR=${BIN_DIR:-/usr/lib/sqoop/bin}
+CONF_DIR=/etc/sqoop/
+CONF_DIST_DIR=/etc/sqoop/conf/
+ETC_DIR=${ETC_DIR:-/etc/sqoop}
+MAN_DIR=${MAN_DIR:-/usr/share/man/man1}
+
+
+
+install -d -m 0755 ${PREFIX}/${LIB_DIR}
+
+install -d -m 0755 ${PREFIX}/${LIB_DIR}
+cp sqoop*.jar ${PREFIX}/${LIB_DIR}
+
+install -d -m 0755 ${PREFIX}/${LIB_DIR}/lib
+cp -a lib/*.jar ${PREFIX}/${LIB_DIR}/lib
+
+#install -d -m 0755 ${PREFIX}/${LIB_DIR}/shims
+#cp -a shims/*.jar ${PREFIX}/${LIB_DIR}/shims
+
+install -d -m 0755 $PREFIX/usr/bin
+
+install -d -m 0755 $PREFIX/${BIN_DIR}
+cp bin/* $PREFIX/${BIN_DIR}
+
+install -d -m 0755 $PREFIX/${DOC_DIR}
+cp docs/*.html  $PREFIX/${DOC_DIR}
+cp docs/*.css $PREFIX/${DOC_DIR}
+cp -r docs/api $PREFIX/${DOC_DIR}
+cp -r docs/images $PREFIX/${DOC_DIR}
+
+
+install -d -m 0755 $PREFIX/$MAN_DIR
+for i in sqoop sqoop-codegen sqoop-export sqoop-import-all-tables sqoop-version sqoop-create-hive-table sqoop-help sqoop-list-databases sqoop-eval sqoop-import sqoop-list-tables sqoop-job sqoop-metastore sqoop-merge
+	do echo "Copying manpage $i"
+	cp docs/man/$i* $PREFIX/$MAN_DIR
+	echo "Creating wrapper for $i"
+	wrapper=$PREFIX/usr/bin/$i
+	mkdir -p `dirname $wrapper`
+	cat > $wrapper <<EOF
+#!/bin/sh
+export HADOOP_HOME=/usr/lib/hadoop
+export SQOOP_HOME=/usr/lib/sqoop
+exec /usr/lib/sqoop/bin/$i "\$@"
+EOF
+   chmod 0755 $wrapper
+done
+
+install -d -m 0755 $PREFIX/$ETC_DIR/conf
+(cd ${BUILD_DIR}/conf && tar cf - .) | (cd $PREFIX/$ETC_DIR/conf && tar xf -)
+rm $PREFIX/$ETC_DIR/conf/.gitignore
+
+unlink $PREFIX/$LIB_DIR/conf || /bin/true
+ln -s /etc/sqoop/conf $PREFIX/$LIB_DIR/conf
diff --git a/cloudera/maven-packaging/pom.xml b/cloudera/maven-packaging/pom.xml
new file mode 100644
index 0000000..b2aaaac
--- /dev/null
+++ b/cloudera/maven-packaging/pom.xml
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<project>
+  <modelVersion>4.0.0</modelVersion>
+  
+  <parent>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>cdh-root</artifactId>
+    <version>3.0-u1-SNAPSHOT</version>
+  </parent>
+  
+  <groupId>com.cloudera.cdh</groupId>
+  <artifactId>sqoop-root</artifactId>
+  <version>3.0-u1-SNAPSHOT</version>
+  <packaging>pom</packaging>
+  
+  <name>CDH Sqoop root</name>
+  
+  <repositories>
+    <repository>
+      <id>cdh.releases.repo</id>
+      <url>https://repository.cloudera.com/content/groups/cdh-releases-rcs</url>
+      <name>CDH Releases Repository</name>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+    <repository>
+      <id>cdh.snapshots.repo</id>
+      <url>https://repository.cloudera.com/content/repositories/snapshots</url>
+      <name>CDH Snapshots Repository</name>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+    </repository>
+  </repositories>
+
+  <modules>
+    <module>../../cloudera-pom.xml</module>
+    <module>sqoop</module>
+  </modules>
+  
+  <dependencies>
+    <dependency>
+      <groupId>com.cloudera.cdh</groupId>
+      <artifactId>hadoop-root</artifactId>
+      <type>pom</type>
+    </dependency>
+    <dependency>
+      <groupId>com.cloudera.cdh</groupId>
+      <artifactId>zookeeper-root</artifactId>
+      <type>pom</type>
+    </dependency>
+
+  </dependencies>
+  
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+          <archive>
+            <manifestFile>target/classes/META-INF/MANIFEST.MF</manifestFile>
+          </archive>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+
diff --git a/cloudera/maven-packaging/sqoop/pom.xml b/cloudera/maven-packaging/sqoop/pom.xml
new file mode 100644
index 0000000..5254bb3
--- /dev/null
+++ b/cloudera/maven-packaging/sqoop/pom.xml
@@ -0,0 +1,149 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+  
+  <parent>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>sqoop-root</artifactId>
+    <version>3.0-u1-SNAPSHOT</version>
+  </parent>
+  
+  <groupId>com.cloudera.sqoop</groupId>
+  <artifactId>sqoop</artifactId>
+  <version>1.2.0-cdh3u1-SNAPSHOT</version>
+  <packaging>pom</packaging>
+  
+  <name>sqoop</name>
+  
+  <dependencies>
+    <dependency>
+      <groupId>com.cloudera.cdh</groupId>
+      <artifactId>sqoop-ant</artifactId>
+      <type>pom</type>
+    </dependency>
+    
+    <dependency>
+      <groupId>commons-cli</groupId>
+      <artifactId>commons-cli</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>commons-collections</groupId>
+      <artifactId>commons-collections</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>commons-lang</groupId>
+      <artifactId>commons-lang</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>commons-io</groupId>
+      <artifactId>commons-io</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>commons-logging</groupId>
+      <artifactId>commons-logging</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-core</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>org.apache.hbase</groupId>
+      <artifactId>hbase</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>hsqldb</groupId>
+      <artifactId>hsqldb</artifactId>
+    </dependency>
+    
+    <dependency>
+      <groupId>ant-contrib</groupId>
+      <artifactId>ant-contrib</artifactId>
+    </dependency>
+    
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-jar</id>
+            <phase>package</phase>
+            <goals>
+              <goal>attach-artifact</goal>
+            </goals>
+            <configuration>
+              <artifacts>
+                <artifact>
+                  <file>${basedir}/../../../build/${project.artifactId}-${project.version}.jar</file>
+                  <type>jar</type>
+                </artifact>
+                <artifact>
+                  <file>${basedir}/../../../build/${project.artifactId}-test-${project.version}.jar</file>
+                  <type>jar</type>
+                  <classifier>tests</classifier>
+                </artifact>
+                <artifact>
+                  <file>${basedir}/target/${project.artifactId}-${project.version}-sources.jar</file>
+                  <type>jar</type>
+                  <classifier>sources</classifier>
+                </artifact>
+              </artifacts>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-source-jar</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <configuration>
+              <target>
+                <mkdir dir="${basedir}/target" />
+                <jar destfile="${basedir}/target/${project.artifactId}-${project.version}-sources.jar">
+                  <fileset dir="${basedir}/../../../src">
+                    <include name="**/*" />
+                  </fileset>
+                </jar>
+              </target>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+  
+</project>
diff --git a/cloudera/maven-packaging/templates/build.template b/cloudera/maven-packaging/templates/build.template
new file mode 100644
index 0000000..4758032
--- /dev/null
+++ b/cloudera/maven-packaging/templates/build.template
@@ -0,0 +1,8 @@
+version=com.cloudera.sqoop:sqoop.version
+sqoop.version=com.cloudera.sqoop:sqoop.version
+reactor.repo=https://repository.cloudera.com/content/repositories/snapshots
+hadoop-core.cloudera.version=org.apache.hadoop:hadoop-core.version
+
+hbase.version=org.apache.hbase:hbase.version
+
+zookeeper.version=org.apache.zookeeper:zookeeper.version
diff --git a/cloudera/maven-packaging/templates/libraries.template b/cloudera/maven-packaging/templates/libraries.template
new file mode 100644
index 0000000..3597d1e
--- /dev/null
+++ b/cloudera/maven-packaging/templates/libraries.template
@@ -0,0 +1,47 @@
+#   Licensed to Cloudera, Inc. under one or more
+#   contributor license agreements.  See the NOTICE file distributed with
+#   this work for additional information regarding copyright ownership.
+#   Cloudera, Inc. licenses this file to You under the Apache License, Version 2.0
+#   (the "License"); you may not use this file except in compliance with
+#   the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+# This properties file lists the versions of the various artifacts we use.
+# It drives ivy and the generation of a maven POM
+
+avro.version=1.5.1
+
+#checkstyle.version=checkstyle:checkstyle.version
+checkstyle.version=5.0
+
+commons-cli.version=commons-cli:commons-cli.version
+commons-collections.version=commons-collections:commons-collections.version
+commons-io.version=commons-io:commons-io.version
+commons-lang.version=commons-lang:commons-lang.version
+commons-logging.version=commons-logging:commons-logging.version
+
+# Cloudera Distribution dependency version
+hadoop-core.cloudera.version=org.apache.hadoop:hadoop-core.version
+
+hbase.version=org.apache.hbase:hbase.version
+
+zookeeper.version=org.apache.zookeeper:zookeeper.version
+
+hsqldb.version=hsqldb:hsqldb.version
+
+ivy.version=org.apache.ivy:ivy.version
+
+junit.version=junit:junit.version
+
+log4j.version=log4j:log4j.version
+
+mvn.version=2.0.10
+
+rats-lib.version=com.google.code.p.arat:rat-lib.version
diff --git a/ivy.xml b/ivy.xml
index b6c4830..1b6dd1f 100644
--- a/ivy.xml
+++ b/ivy.xml
@@ -18,6 +18,7 @@
 <ivy-module version="2.0" xmlns:m="http://ant.apache.org/ivy/maven"
             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
             xsi:noNamespaceSchemaLocation="http://ant.apache.org/ivy/schemas/ivy.xsd">
+
   <info organisation="com.cloudera.sqoop" module="${name}">
     <license name="Apache 2.0"/>
     <ivyauthor name="Sqoop Team" url="http://github.com/cloudera/sqoop" />
@@ -65,12 +66,23 @@
     <artifact conf="master"/>
   </publications>
   <dependencies>
+
     <!-- Dependencies for Cloudera's Distribution for Hadoop -->
-    <dependency org="com.cloudera.hadoop" name="hadoop-core"
+    <dependency org="org.apache.hadoop" name="hadoop-core"
       rev="${hadoop-core.cloudera.version}" conf="cloudera->default"/>
-    <dependency org="com.cloudera.hadoop" name="hadoop-test"
+    <dependency org="org.apache.hadoop" name="hadoop-test"
       rev="${hadoop-core.cloudera.version}" conf="clouderatest->default"/>
 
+      <dependency org="org.apache.hbase" name="hbase"
+        rev="${hbase.version}" conf="cloudera->default">
+          <artifact name="hbase" type="jar"/>
+          <artifact name="hbase" type="test-jar" ext="jar" m:classifier="tests"/>
+          <exclude org="com.sun.jersey" module="jersey-core"/>
+          <exclude org="com.sun.jersey" module="jersey-json"/>
+          <exclude org="com.sun.jersey" module="jersey-server"/>
+          <exclude org="org.apache.thrift" module="thrift"/>
+      </dependency>
+
     <!-- Common dependencies for Sqoop -->
     <dependency org="commons-cli" name="commons-cli"
       rev="${commons-cli.version}" conf="common->default"/>
diff --git a/ivy/ivysettings.xml b/ivy/ivysettings.xml
index ce4c522..8dc2183 100644
--- a/ivy/ivysettings.xml
+++ b/ivy/ivysettings.xml
@@ -38,7 +38,10 @@
       value="https://repository.cloudera.com/content/repositories/staging/"
       override="false"/>
   <property name="releases.cloudera.com"
-      value="https://repository.cloudera.com/content/repositories/releases/"
+      value="https://repository.cloudera.com/content/groups/cdh-releases-rcs/"
+      override="false"/>
+  <property name="snapshots.cloudera.com"
+      value="https://repository.cloudera.com/content/repositories/snapshots/"
       override="false"/>
   <property name="maven2.pattern"
       value="[organisation]/[module]/[revision]/[artifact]-[revision](-[classifier])"/>
@@ -48,13 +51,22 @@
   <settings defaultResolver="${resolvers}"/>
 
   <resolvers>
+      <!-- CDH build repo, the pattern attribute handles SNAPSHOT artifacts -->
+      <ibiblio name="reactor-repo" root="${reactor.repo}"
+          m2compatible="true"
+          checkmodified="true"
+          pattern="[organisation]/[module]/[revision]/[artifact]-[revision](-[classifier]).[ext]"/>
+
     <ibiblio name="maven2" root="${repo.maven.org}"
         pattern="${maven2.pattern.ext}" m2compatible="true"/>
     <ibiblio name="cloudera-releases" root="${releases.cloudera.com}"
         pattern="${maven2.pattern.ext}" m2compatible="true"/>
     <ibiblio name="apache-snapshot" root="${snapshot.apache.org}"
+        m2compatible="true" checkmodified="true"
+        pattern="[organisation]/[module]/[revision]/[artifact]-[revision](-[classifier]).[ext]"/>
+    <ibiblio name="cloudera-releases" root="${releases.cloudera.com}"
         m2compatible="true" checkmodified="true" changingPattern=".*SNAPSHOT"/>
-    <ibiblio name="cloudera-staging" root="${staging.cloudera.com}"
+    <ibiblio name="cloudera-snapshots" root="${snapshots.cloudera.com}"
         m2compatible="true" checkmodified="true" changingPattern=".*SNAPSHOT"/>
 
     <filesystem name="fs" m2compatible="true">
@@ -64,16 +76,18 @@
 
     <chain name="default" dual="true" checkmodified="true"
         changingPattern=".*SNAPSHOT">
-      <resolver ref="apache-snapshot"/> 
+      <resolver ref="reactor-repo"/>
+      <resolver ref="apache-snapshot"/>
       <resolver ref="cloudera-releases"/>
-      <resolver ref="cloudera-staging"/> 
+      <resolver ref="cloudera-snapshots" />
       <resolver ref="maven2"/>
     </chain>
 
     <chain name="internal" dual="true">
+      <resolver ref="reactor-repo"/>
       <resolver ref="fs"/>
       <resolver ref="apache-snapshot"/> 
-      <resolver ref="cloudera-staging"/> 
+      <resolver ref="cloudera-releases"/>
       <resolver ref="maven2"/>
     </chain>
 
-- 
1.7.0.4

